src/UI/Art/SpriteArtist/SpriteLibrary.java

```diff
@@ -314,24 +314,24 @@ public static SpriteSetKey instance(TerrainType terrain, Commander co)
 
   private static class UnitSpriteSetKey
   {
-    public final UnitModel.UnitEnum unitTypeKey;
+    public final String unitTypeKey;
     public final Faction factionKey;
     public final Color colorKey;
     private static ArrayList<UnitSpriteSetKey> instances = new ArrayList<UnitSpriteSetKey>();
 
-    private UnitSpriteSetKey(UnitModel.UnitEnum unitType, Faction faction, Color color)
+    private UnitSpriteSetKey(String unitType, Faction faction, Color color)
     {
       unitTypeKey = unitType;
       factionKey = faction;
       colorKey = color;
     }
 
-    public static UnitSpriteSetKey instance(UnitModel.UnitEnum unitType, Faction faction, Color color)
+    public static UnitSpriteSetKey instance(String unitType, Faction faction, Color color)
     {
       UnitSpriteSetKey key = null;
       for( int i = 0; i < instances.size(); ++i )
       {
-        if( instances.get(i).unitTypeKey == unitType && instances.get(i).factionKey == faction && instances.get(i).colorKey == color)
+        if( instances.get(i).unitTypeKey.equalsIgnoreCase(unitType) && instances.get(i).factionKey == faction && instances.get(i).colorKey == color)
```

Comment:
Do you not need run `unitType` through `standardizeID` to ensure we are comparing equal things?